src/components/Place/PlaceDetails/PlaceDetails.tsx

```diff
@@ -25,20 +36,78 @@ export type PlaceDetailsProps = {
 
 export enum PlaceDetailsTab {
   About = "About",
-  Realms = "Realms",
+  ContentModeration = "Content Moderation",
 }
 
 export default React.memo(function PlaceDetails(props: PlaceDetailsProps) {
   const { place, loading } = props
   const l = useFormatMessage()
   const [pois] = useAsyncMemo(getPois)
   const [activeTab, setActiveTab] = useState(PlaceDetailsTab.About)
+  const [openContentModerationModal, setOpenContentModerationModal] =
+    useState(false)
+  const [openConfirmModal, setOpenConfirmModal] = useState(false)
+  const [selectedRate, setSelectedRate] = useState<
+    SceneContentRating | boolean
+  >(false)
+
+  const [ff] = useFeatureFlagContext()
+  const [account] = useAuthContext()
+  const admin = isAdmin(account)
+
+  const rating = useMemo(
+    () => place?.content_rating?.toLocaleUpperCase() || SceneContentRating.TEEN,
+    [place]
+  )
 
   const isPoi = useMemo(
     () => intersects(place?.positions || [], pois || []),
     [place, pois]
   )
 
+  const handleRatingButton = useCallback(
+    (e: React.MouseEvent<any>, rating: SceneContentRating) => {
+      e.stopPropagation()
+      e.preventDefault()
+      setSelectedRate(rating)
+      const data = new Set(Object.values(SceneContentRating))
+      const placeRating = data.has(place.content_rating)
+        ? place.content_rating
+        : SceneContentRating.TEEN
+      if (placeRating !== rating) {
+        setOpenContentModerationModal(true)
+      }
+    },
+    [place, setSelectedRate, setOpenContentModerationModal]
+  )
+
+  const handleChangeRating = useCallback(
+    async (e: React.MouseEvent<any>) => {
+      e.stopPropagation()
+      e.preventDefault()
+      if (typeof selectedRate === "boolean") return
+
+      setOpenContentModerationModal(false)
+
+      await Places.get().updateRating(place.id, {
+        content_rating: selectedRate as SceneContentRating,
+      })
+
+      setOpenConfirmModal(true)
+    },
+    [place, selectedRate, setOpenContentModerationModal, setOpenConfirmModal]
+  )
+
+  const handleConfirmRating = useCallback(
```

Comment:
Better try to use an `onChange` an let the Main component handle the connection with the Server and manage the modals